examples/sol-contract/src/processor.rs

```diff
@@ -0,0 +1,110 @@
+//! Program instruction processor
+//! Only the program admin can issue the Init instruction.
+//! And anyone can check the loan with the Loan2Value instruction.
+
+use solana_program::account_info::{
+    next_account_info,
+    AccountInfo,
+};
+use solana_program::entrypoint::ProgramResult;
+use solana_program::msg;
+use solana_program::program_error::ProgramError;
+use solana_program::program_pack::{
+    IsInitialized,
+    Pack,
+};
+use solana_program::pubkey::Pubkey;
+
+use borsh::BorshDeserialize;
+use pyth_sdk_solana::load_price_feed_from_account_info;
+
+use crate::instruction::ExampleInstructions;
+use crate::state::AdminConfig;
+
+pub fn process_instruction(
+    _program_id: &Pubkey,
+    _accounts: &[AccountInfo],
+    input: &[u8],
+) -> ProgramResult {
+    let account_iter = &mut _accounts.iter();
+    let signer = next_account_info(account_iter)?;
+    let data_account = next_account_info(account_iter)?;
+    let pyth_loan_account = next_account_info(account_iter)?;
+    let pyth_collateral_account = next_account_info(account_iter)?;
+
+    let instruction = ExampleInstructions::try_from_slice(input)?;
+    match instruction {
+        ExampleInstructions::Init {} => {
+            if !(signer.key == _program_id && signer.is_signer) {
+                return Err(ProgramError::Custom(0));
+            }
+
+            let mut loan_info = AdminConfig::unpack_from_slice(&data_account.try_borrow_data()?)?;
+
+            if loan_info.is_initialized() {
+                return Err(ProgramError::Custom(1));
+            }
+
+            loan_info.is_initialized = true;
+            loan_info.loan_price_feed_id = *pyth_loan_account.key;
```

Comment:
If you only need the key there is no need to pass the account (and it can be passed as data). 

However I think it's good to keep the accounts passing and make sure they are a Pyth account. You can try to parse them into a price feed and make sure it's ok.